autocomplete.js

```diff
@@ -48,6 +50,31 @@ async function listRegions(query, pluginSettings, actionParams) {
   });
 }
 
+async function listRoles(query, pluginSettings, actionParams) {
+  const { settings, params } = paramsMapper(pluginSettings, actionParams);
+  const e = getIAM(params, settings);
+  const roles = await e.listRoles().promise().catch((err) => {
+    console.error(err);
+    throw new Error(MISSING_OR_INCORRECT_CREDENTIALS_ERROR);
+  });
+
+  return roles.Roles
+    .filter(({ RoleName, AssumeRolePolicyDocument }) => {
+      let policy;
+      try {
+        policy = JSON.parse(decodeURIComponent(AssumeRolePolicyDocument));
+      } catch {
+        throw new Error(CORRUPTED_ROLE_POLICY_TEXT(RoleName));
+      }
+      const roleIntendedForEks = policy.Statement.some(
+        (item) => item.Principal.Service.includes(EKS_SERVICE_URL),
+      );
+      return RoleName.includes(query) && roleIntendedForEks;
```

Comment:
That's pretty long (and logic infused) body for a `filter` callback. Please consider extracting validation code to a separate function in helpers.js and then applying this function in `filter` instead.